liquibase-core/src/main/java/liquibase/resource/ResourceAccessor.java

```diff
@@ -257,4 +308,51 @@ public OutputStream openOutputStream(OpenOptions openOptions) throws IOException
             return openOutputStream(new OpenOptions());
         }
     }
+
+    class SearchOptions {
+        private int minDepth;
+        private int maxDepth;
+
+        public SearchOptions() {
+            minDepth = 0;
+            maxDepth = 1;
```

Comment:
Should not the default value be  Integer.MAX_VALUE ? This way it will behave the same as before if new people just do `new SearchOptions() `

Reply:
In my opinion, it wouldn't change that much when SearchOptions is initialized it will be followed up with min/maxDepth seters. It we do initialize it with `Integer.MAX_VALUE` it will still define whether it's a recursive search depending on the values set. To keep recursive compatibility as it was before this change, there is a `search()` which expect a `recursive` boolean flag instead of a SearchOptions (as well as an IncludeAll method which does not expect min/max values which will end up performing a recursive search `min=0` - `max=Integer.MAX_VALUE`)